src/client.rs

```diff
@@ -0,0 +1,41 @@
+use crate::error::Error;
+use prost_types::Any;
+
+pub struct Client(internal::client::DaprClient<tonic::transport::Channel>);
```

Comment:
I suggest to use something like `Client<T>(internal::client::DaprClient<T>)`, it more easy to write testing with mock library.

Reply:
Hmmm... perhaps we could do a default type parameter to `tonic::transport::Channel`. I would really prefer to not leak the internal transport implementation...

Reply:
I think we have a little different understanding of the internal types of leaks.

I understand you want all 3rd party types are hidden behide SDK types, so, user can only see `dapr::Client` instead of `dapr::Client<tonic::transport::Channel>` in document. 

But my understanding is that we only need to make sure that the functions and types that all users will use directly are not limited to a specific implementation.

For example, `dapr::connect` will return `dapr::Client<T>`, user does not care what type of T is returned, they just use it to `invoke_service`, or use high level [generated stub](https://github.com/flier/rust-dapr/blob/master/dapr/examples/caller.rs#L18).

If we provide a mocking implementation, it only need replace with `dapr::mock::connect` which returns `dapr::Client<dapr::MockChannel>`, and all other code should be same. Or if we provide a layered tracing implementation, there should be a `tracing` future and `dapr::connect` returns something like `dapr::Client<Tracing<tonic::transport::Channel>>`

Reply:
If we don't choose which type `T`, the user will have to choose which means either we choose for the user or the user has to know about `tonic`.

Reply:
Ok, I think `Client` is good enough for the Scaffold :)